### ISSUE TITLE
The issue about step-1

### ISSUE THREAD

**Issue by Xtian-hub** [WeixiaoGao/PSSNet#9]
I have completed the software construction, but after running it, there have been no changes to the source files and no new files generated !
![4](https://github.com/tudelft3d/PSSNet/assets/116425364/be12ecd5-4935-4f2e-b7c4-f79ce4a58d58)
 I followed the file format of the ‘data_demo_pssnet’, for example, 1.png.![1](https://github.com/tudelft3d/PSSNet/assets/116425364/91cd72b4-25e9-4d9f-8712-5f2faba6d5a8). The '2.png' is my folder structure
![2](https://github.com/tudelft3d/PSSNet/assets/116425364/6911d122-4cec-44d2-9a1d-e80d73527b4a).  Furthermore, I found an example in data_demo_pssnet. Should the format of the config.txt I need to write be similar to that file? Below is the config.txt I placed.
[config.txt](https://github.com/tudelft3d/PSSNet/files/14782991/config.txt) Regarding the content of config.txt, how should I set the data source path and the output path? As I understand it, root_path is the data source path where I placed *.ply and *.jpg files for testing, but there was no response. Is it correct for me to do this?
![1](https://github.com/tudelft3d/PSSNet/assets/116425364/574a610e-e91b-4c5e-a55f-f241ccc732a8)
Thank you for your time and assistance. I sincerely hope to overcome this difficulty and begin making full use of your software.

Best regards,

Geng Xiaotian

**Comment by Xtian-hub** [comment]
sorry, I change like "5.png" ![5](https://github.com/tudelft3d/PSSNet/assets/116425364/02c189d5-aca9-4759-8948-4e5a3cd2786a)

**Comment by WeixiaoGao** [comment]
Hi, 

The error indicates an issue with the dataset's root path specified in config.txt; it should be correctly set to ```root_path = E:/openlibs/datasets/```. Additionally, your data should be placed inside the ```input/ ``` folder (train/test/validate/predict), not at the root path. Ensure that you either create or download all necessary folders, following the folder structure outlined in the demo. While you can adjust settings in 'config.txt', remember to do so in accordance with the guidelines provided for each parameter.

**Comment by Xtian-hub** [comment]
您好，我是耿晓天我已经收到了您的来信，稍后我会回复，谢谢?7?8(?1?1ω?1?1)?1?7

**Comment by Xtian-hub** [comment]
Hi, Dr.Gao
My program has run successfully. I want to achieve the results of over-segmentation. but segmentation results are not appearing
![12](https://github.com/tudelft3d/PSSNet/assets/116425364/b7cd8699-7363-450c-b592-073382660a99)
. I encountered this in the code.
![13](https://github.com/tudelft3d/PSSNet/assets/116425364/2308a247-a641-45bf-ada5-fca89b7b07fb)
 I have already created the folders according to the demo data. I changed the operating_mode in this order
![14](https://github.com/tudelft3d/PSSNet/assets/116425364/c4957f38-e273-4585-8a3e-d7e4b00f6854)
. Can you please advise me on how to resolve this?

**Comment by WeixiaoGao** [comment]
It appears that the output for the test dataset from the previous step (planar and non-planar classification) may be incorrect. Could you please verify that the folder designated for saving the test data exists and that the save path matches precisely with the one mentioned in the error message? After confirming these details, it will be necessary to rerun the pipeline for the test dataset.

**Comment by Xtian-hub** [comment]
Hi, Dr.Gao
the issue still persist.
(py38) E:\openlibs\data_demo_pssnet\PSSNet\step-1\build-vs\semantic_urban_mesh_segmentation\Release>semantic_urban_mesh_segmentation.exe E:\openlibs\data_demo_pssnet\

Reading configuration file: Done in (s): 0.0007669

--------------------- PSSNet oversegmented train mesh ---------------------
        Get all training data.
        Done in (s): 0.0018128

        - Planar and non-planar based region growing.
[PropertyContainer] A property with name "f:label_probabilities" already exists. Returning invalid property.
loading training mesh: E:/openlibs/data_demo_pssnet/PSSNet/input/train/Tile_+1984_+2688_groundtruth_L0.ply
        Loading semantic mesh: Tile_+1984_+2688
mesh read successful, loading time: 2.00983s
  The total number of input triangle facets:  309445

  The total number of input edges:  466157

  The total number of input vertices:  156580

  The total number of input texture image:  1

Start to reading sampled pointcloud Tile_+1984_+2688
Compute initial vertex planarity for vertex
 Done in (s): 1.93838

  Region growing, use  (s): 5.04177

The number of intermediate segments are 9016
  assign_initial_segments_on_mesh, use  Global smoothed 13296 faces.
 (s): 53.5626

 After region growing, the segment number is 4858
The number of segments are 4858
        Saving segment mesh: Tile_+1984_+2688
TODO: use binary format
mesh saved
        Done in (s): 14.4375

--------------------- PSSNet oversegmented test mesh  ---------------------
        Get all test data.
        Done in (s): 0.0007713

        - Planar and non-planar based region growing.
[PropertyContainer] A property with name "f:label_probabilities" already exists. Returning invalid property.
loading training mesh: E:/openlibs/data_demo_pssnet/PSSNet/input/test/Tile_+1984_+2688_groundtruth_L0.ply
        Loading semantic mesh: Tile_+1984_+2688
RPly: Unable to open file
failed to open ply file 'E:/openlibs/data_demo_pssnet/PSSNet/output/test/Tile_+1984_+2688_mesh_classification.ply'
reading file failed (no data exist)
File loading failed
I have made sure that the file exists, but the data is not generated. but data not generation. I cannot determine whether the issue is related to 'f:label_probabilities', i can not find it.
![3](https://github.com/tudelft3d/PSSNet/assets/116425364/ab7e2d51-9536-45ad-91fc-a8e3684cfa81)
I executed the process in this order: "PSSNet_pipeline_for_GCN: Run the following modes in sequence: Get_labels_for_planar_non_planar_from_semantic, Pipeline, PSSNet_oversegmentation, PSSNet_oversegmentation_evaluation, PSSNet_graph_construction, and PSSNet_pcl_generation_for_GCN."

**Comment by Xtian-hub** [comment]
sorry, I got another error.
![5](https://github.com/tudelft3d/PSSNet/assets/116425364/5530742d-42d6-4c53-92b7-c4a03a647f7e)

**Comment by WeixiaoGao** [comment]
The f:label_probabilities field is not crucial for our current issue. Could you please visualize the data located at E:/openlibs/data_demo_pssnet/PSSNet/input/test/Tile_+1984_+2688_groundtruth_L0.ply, as indicated in the error message?

The PSSNet_pipeline_for_GCN encompasses all necessary modes in a sequence, eliminating the need to run them individually under normal circumstances.

Regarding the encodings issue, this is a standard Python package. It appears there might be a problem with your Python environment variable configuration. Could you verify its setup?

**Comment by Xtian-hub** [comment]
the data located at ‘E:/openlibs/data_demo_pssnet/PSSNet/input/test/Tile_+1984_+2688_groundtruth_L0.ply’ as follow: ![1](https://github.com/tudelft3d/PSSNet/assets/116425364/c5055dda-0010-44c8-b13b-87ff904354fb)
![2](https://github.com/tudelft3d/PSSNet/assets/116425364/66bc293b-cb5c-434c-b12e-53016a9009d7).it do not generation TextureFile.

**Comment by WeixiaoGao** [comment]
I'm referring to the results of the planar and non-planar classification, which are expected to be located in the E:/openlibs/data_demo_pssnet/PSSNet/output/test/ directory. These should be produced automatically by the PSSNet_pipeline_for_GCN mode. If they're missing, could you please rerun the pipeline in 'Pipeline' mode to regenerate the classification results for the test data?

**Comment by Berlin000000** [comment]
Regarding the handling of step 1, I also have some questions. If I want to directly generate new data processing results in the form of predict/test, how should I set it up? When I set the operating_mode to PSSNet_pipeline_for_GCN, the software reminds me of the issue that there is no data or data labels in the train directory. How should I place my test data?

**Comment by Xtian-hub** [comment]
您好，我是耿晓天我已经收到了您的来信，稍后我会回复，谢谢?7?8(?1?1ω?1?1)?1?7

**Comment by Berlin000000** [comment]
For example, here when I set the operating_mode to PSSNet_pipeline_for_GCN and place the data in the \context\PSSNet\input\test directory,
 I encounter an error while processing Get_labels_for_planar_non_planar_from_semantic.

![Snipaste_2024-04-21_15-10-34](https://github.com/tudelft3d/PSSNet/assets/48549786/0e858c5a-648d-41ce-ad64-835495309f34)

**Comment by WeixiaoGao** [comment]
The `test` folder should contain the ground truth labels for evaluation. If it does not, please place your data in the `predict` folder instead.

**Comment by Berlin000000** [comment]
Thank you for your help. I have set the mode to "predict." When I run operating_mode = PSSNet_oversegmentation, I find that it requires reading the pnp information of the model, which is not present in the original data. Could it be that there is still an issue with the parameters in my configuration?
![image](https://github.com/tudelft3d/PSSNet/assets/48549786/a3584b7f-0fbd-4f6a-9ad9-a73fc65ebd1e)
![image](https://github.com/tudelft3d/PSSNet/assets/48549786/b21e4aa6-250d-4b73-8de3-2b4f5139f961)
![image](https://github.com/tudelft3d/PSSNet/assets/48549786/52baea0e-9998-4de1-99b3-7d5747844748)
